pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/kubernetes/KubernetesRuntimeFactory.java

```diff
@@ -280,6 +280,8 @@ public KubernetesRuntime createContainer(InstanceConfig instanceConfig, String c
                 instanceFile = pythonInstanceFile;
                 break;
             case GO:
+                // Go functions should communicate via grpcPort directly.
+                instanceConfig.setPort(grpcPort);
```

Comment:
I agree that these added lines fix the problem, but I think the following might be a more general solution:

```
        // pass grpcPort configured in functionRuntimeFactoryConfigs.grpcPort in functions_worker.yml
        if (grpcPort != null) {
            instanceConfig.setPort(grpcPort);
        }
```

I would add this down where the `metricsPort` changes are made on line 305 in the original file.

My primary motivation for updating the `instanceConfig` for all runtimes is that the port configuration is currently _wrong_ for all runtimes. The Java and Python runtimes are not broken because they get the port value in another part of the code base. I would prefer a general solution because it ensures that the `instanceConfig` port is the right value for every runtime, even if it isn't used.

Reply:
Yeah, I was being too conservative. This is the right call.